ciphers/xor_cipher.py

```diff
@@ -29,12 +28,14 @@ def __init__(self, key: int = 0):
         # private field
         self.__key = key
 
-    def encrypt(self, content: str, key: int) -> list[str]:
+    def encrypt(self, content: str, key: int, type_: int) -> list[str] | str:
```

Comment:
Why don't you overload and typehint `type_` to either literal `0` or literal `1`